clients/go/README.md

```diff
@@ -1,63 +1,93 @@
-You can port over a Go Ethereum application by using a `sapphire.WrappedBackend`
-or by packing native Ethereum transactions Sapphire style.
+[@oasisprotocol/sapphire-paratime] makes it easy to port your dapp to the
+[Sapphire ParaTime]. You can port over a Go Ethereum application by using a
+`sapphire.WrappedBackend` or by packing native Ethereum transactions Sapphire
+style.
 
 [@oasisprotocol/sapphire-paratime]: https://pkg.go.dev/github.com/oasisprotocol/sapphire-paratime/go/
-[sapphire paratime]: https://docs.oasis.io/general/developer-resources/sapphire-paratime/
-
-_If your dapp doesn't port in under 10 minutes, it's a bug!_  
-
-If you have more than a little trouble, please file an issue. There should be
-_no_ reason _not_ to use the Sapphire ParaTime!
+[Sapphire ParaTime]: https://docs.oasis.io/dapp/sapphire
 
 ## Building
 
-Sapphire Paratime compatibility library works with `Go` version 1.17 or later and the latest comparable `go-ethereum` version.
+Sapphire compatibility library works with Go version 1.22 or later and the
+latest comparable `go-ethereum` version.
 
-To build and test locally, import the `Go` package.
+To build and test locally:
+
+```shell
+go test
+```
 
 ## Usage
 
+### Import
+
+```go
+import (
+    "context"
+
+    "github.com/ethereum/go-ethereum/accounts/abi/bind"
+    "github.com/ethereum/go-ethereum/ethclient"
+
+    sapphire "github.com/oasisprotocol/sapphire-paratime/clients/go"
+)
+```
+
 ### Go-Ethereum ABI
 
 After [generating](https://geth.ethereum.org/docs/dapp/abigen) the Go bindings
-for a particular Solidity contract, you can instantiate an Ethereum client with the
-Sapphire Paratime gateway URL and instantiate a `sapphire.WrappedBackend` as a drop in
-replacement.
+for a particular Solidity contract, you can instantiate an Ethereum client with
+the Sapphire gateway URL and instantiate a `sapphire.WrappedBackend` as a drop
+in replacement:
 
-```Go
+```go
 // key := private key
-c, _ := ethclient.Dial(sapphire.Networks[SapphireChainID.Uint64()].DefaultGateway)
-backend := sapphire.WrapClient(*c, func(digest [32]byte)([]byte, error) {
+client, _ := ethclient.Dial(sapphire.Networks[SapphireChainID.Uint64()].DefaultGateway)
+backend, _ := sapphire.WrapClient(client, func(digest [32]byte)([]byte, error) {
   // Pass in a custom signing function to interact with the signer
   return crypto.Sign(digest[:], key)
 })
 ```
 
-Contracts using `go-ethereum`'s `abigen` can be used by passing in `backend` instead of the usual `ethclient`. For example,
+Contracts using `go-ethereum`'s `abigen` can now be used by passing in `backend`
+instead of the usual `ethclient.Client` instance:
 
-```Go
-txOpts := backend.Transactor(senderAddr)
+```go
 nft, _ := NewNft(addr, backend)
+```
+
+Confidential transactions using Go-Ethereum ABI wrapper can be submitted by
+passing the Sapphire-specific `bind.TransactOpts` as the first parameter:
```

Comment:
Maybe add a warning that in case this transactor is not passed, transactions will be plain-text?